From a89e3cf8fe96192f216de6718528b2b331920579 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Mon, 29 Jul 2024 16:24:11 +0200
Subject: [PATCH 1071/1695] dts: stm32: add iwdg2 node on stm32mp23 platform
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add support of IWDG2 peripheral on MP23.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a0507ce1fb740692fb64049a4160e55978acaa16 dts: stm32: add iwdg2 node on stm32mp23 platform]
Change-Id: Ia26a7da4d78c08a21f54eed96a1023b751ef2c58
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/395618
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
Tested-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index a4ee0bae7..562b43e5a 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -459,6 +459,16 @@
 				status = "disabled";
 			};
 
+			iwdg2: watchdog@44020000 {
+				compatible = "st,stm32mp1-iwdg";
+				reg = <0x44020000 0x400>;
+				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_IWDG2>, <&rcc LSI_CK>;
+				clock-names = "pclk", "lsi";
+				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG2_ID>;
+				status = "disabled";
+			};
+
 			lptimer3: timer@46050000 {
 				compatible = "st,stm32mp25-lptimer";
 				reg = <0x46050000 0x400>;
-- 
2.47.2

